packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -171,7 +171,7 @@ export function processSyncBlocks(
     blockBuffer = sortBlocks(blockBuffer);
     while(blockBuffer.length > 0) {
       const block = blockBuffer.shift();
-      if(config.types.Root.equals(headRoot, block.message.parentRoot)) {
+      if(!trusted || (trusted && config.types.Root.equals(headRoot, block.message.parentRoot))) {
```

Comment:
for regular sync, we should not check for the `headRoot` since we're not sure block `slot` is parent of block `slot + 1`

Reply:
I think for initial sync we should also check if received block slot is greater than head block slot.

Also as a backup, if we get  `Received block parent root doesn't match our head` we should set lastBlockTarget to current head slot, so we try to refetch that range in case some peer sent response with some missing block

Reply:
We can't use `trusted` to determine if this is used in initial vs regular sync. Bc we need to be running with trusted=false during initial sync too.